src/amber/controller/base.cr

```diff
@@ -1,16 +1,23 @@
 require "http"
+require "./*"
 
 module Amber::Controller
   class Base
-    property request : HTTP::Request?
-    property response : HTTP::Server::Response?
-    property raw_params : HTTP::Params?
-    property context : HTTP::Server::Context?
+    include Redirect
 
-    protected def set_context(@context : HTTP::Server::Context)
-      self.request = context.request
-      self.response = context.response
-      self.raw_params = context.params
+    getter _context : HTTP::Server::Context?
+
+    getter _request : HTTP::Request = HTTP::Request.new("GET", "/")
+    getter _response : HTTP::Server::Response =  HTTP::Server::Response.new(IO::Memory.new)
+    getter _params : HTTP::Params = HTTP::Params.parse("")
+    getter params : Amber::Validators::Params?
+
+    def _set_context(@_context : HTTP::Server::Context)
```

Comment:
Would it be possible to set all of this with `def initialize` do keep it more standard?